airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -3021,7 +3021,15 @@ def execute(self, context: Any):
         # Wait for the job to complete
         if not self.deferrable:
             job.result(timeout=self.result_timeout, retry=self.result_retry)
-            self._handle_job_error(job)
+            while True:
+                if job.state in ("PENDING", "RUNNING"):
+                    import time
+
+                    time.sleep(5)
```

Comment:
I don't think we need to add a sleep here. Why not do...

```python
while job.state in ("PENDING", "RUNNING"):
...
```

Reply:
@RNHTTR @kandharvishnu In a second thought, maybe it's better to have a `time.sleep()` - otherwise, if it is pending or running for long time - requests will be sent immedately, which might cause exceeding API rate limit.
Example from another Google providers:

https://github.com/apache/airflow/blob/3c58e01266f884544fdebc70f92b63848c610d2d/providers/src/airflow/providers/google/cloud/hooks/dataproc.py#L1241

What do you guys think?

Reply:
@pankajastro , fyi for #44279

Reply:
Yes, if we are looping, then sleep makes sense.